code/modules/vehicles/armored/som_armored_weapons.dm

```diff
@@ -0,0 +1,258 @@
+	///Current ammo override to use based on power level
+	var/current_ammo_type = /datum/ammo/rocket/coilgun
+	///Power setting toggle action
+	var/datum/action/item_action/coilgun_power/power_toggle
+
+/obj/item/armored_weapon/coilgun/Initialize(mapload)
+	. = ..()
+	power_toggle = new(src)
+
+/obj/item/armored_weapon/coilgun/Destroy()
+	QDEL_NULL(power_toggle)
+	return ..()
+
+/obj/item/armored_weapon/coilgun/attach(obj/vehicle/sealed/armored/tank, attach_primary)
+	. = ..()
+	RegisterSignal(tank, COMSIG_VEHICLE_GRANT_CONTROL_FLAG, PROC_REF(give_action))
+
+/obj/item/armored_weapon/coilgun/detach(atom/moveto)
+	UnregisterSignal(chassis, COMSIG_VEHICLE_GRANT_CONTROL_FLAG)
+	. = ..()
+
+/obj/item/armored_weapon/coilgun/do_fire(turf/source_turf, ammo_override)
+	ammo_override = current_ammo_type
+	var/x_offset = 0
+	var/y_offset = 0
+	var/animation_duration = 0.9 SECONDS
+	switch(power_level)
+		if(COILGUN_MED_POWER)
+			switch(chassis.dir)
+				if(NORTH)
+					y_offset = -15
+				if(SOUTH)
+					y_offset = 15
+				if(EAST)
+					x_offset = -15
+				if(WEST)
+					x_offset = 15
+		if(COILGUN_HIGH_POWER)
+			animation_duration = 1.2 SECONDS
+			switch(chassis.dir)
+				if(NORTH)
+					y_offset = -25
+				if(SOUTH)
+					y_offset = 25
+				if(EAST)
+					x_offset = -25
+				if(WEST)
+					x_offset = 25
+	if(x_offset || y_offset)
+		animate(chassis, time = 0.3 SECONDS, flags = ANIMATION_RELATIVE|ANIMATION_END_NOW, pixel_x = x_offset, pixel_y = y_offset)
+		animate(time = animation_duration -  0.3 SECONDS, easing = SINE_EASING, flags = ANIMATION_RELATIVE, pixel_x = -x_offset, pixel_y = -y_offset)
+		if(istype(chassis, /obj/vehicle/sealed/armored/multitile/som_tank)) //byond animations are very smelly, there is no way to have these two anims running together nicely
+			addtimer(CALLBACK(chassis, TYPE_PROC_REF(/obj/vehicle/sealed/armored/multitile/som_tank, animate_hover)), animation_duration)
+	return ..()
+
+/obj/item/armored_weapon/coilgun/eject_ammo()
+	for(var/mob/occupant AS in chassis.occupants)
+		occupant.hud_used.update_ammo_hud(src, list(hud_state_empty, hud_state_empty), 0)
+	QDEL_NULL(ammo)
+
+/obj/item/armored_weapon/coilgun/ui_action_click(mob/user, datum/action/item_action/action)
+	var/datum/action/item_action/coilgun_power/power_action = action
+	if(!istype(power_action))
+		return ..()
+	toggle_power_level(user)
+	return TRUE
+
+///Switches between coil power levels
+/obj/item/armored_weapon/coilgun/proc/toggle_power_level(mob/user)
+	power_level += 1
+	if(power_level > COILGUN_HIGH_POWER)
+		power_level = COILGUN_LOW_POWER
+	switch(power_level)
+		if(COILGUN_LOW_POWER)
+			current_ammo_type = /datum/ammo/rocket/coilgun/low
+			windup_delay = 0
+		if(COILGUN_MED_POWER)
+			current_ammo_type = /datum/ammo/rocket/coilgun
+			windup_delay = 0.6 SECONDS
+		if(COILGUN_HIGH_POWER)
+			current_ammo_type = /datum/ammo/rocket/coilgun/high
+			windup_delay = 1 SECONDS
+	to_chat(user, "power level set to [power_level]")
```

Comment:
ballon alert?

Reply:
added

Reply:
this looks really bad, and doesn't need a balloon either tbh. Also the one above is updated, I just can't comment on it or something.